### ISSUE TITLE
When pulled from Medium, the articles do not show the code snippets embedded in them.

### ISSUE THREAD

**Issue by Raahul-Singh** [OpenAstronomy/Universe_OA#53]
For example, in my [blog for SunPy](https://medium.com/@_hawks_/chapter-1-apricity-aef3bd172dab), there are a few code snippets. 
The [same blog](https://openastronomy.org/Universe_OA/posts/2020/06/20200615_0026_raahul-singh/) on OpenAstronomy's student blogs does not show the code snippets.

**Comment by Yash-g17** [comment]
Hey @dpshelio I'd Love to work on this issue

**Comment by Abinash-bit** [comment]
Hey @dpshelio  I have worked on this issue please go through the PR.

**Comment by dpshelio** [comment]
The issue here as provided in #95 (and #96) doesn't solve this problem (it may solve a different formatting problem though). This is to do with the iframe, and how the rss doesn't get the right input. As in this case:

https://github.com/OpenAstronomy/Universe_OA/blob/run/posts/2020/06/20200615_0026_Raahul-Singh.rst?plain=1#L38-L43

**Comment by aditya-pandey-dev** [comment]
Hey @dpshelio , I’d  like  to work on this issue! Here’s how I’m thinking of approaching the problem:

First, I’ll go over the previous comments, related links, and the diagnostics to get the full picture. The main problem seems to be that code snippets (iframes or code blocks) from Medium blogs are not getting parsed or displayed correctly when pulled via RSS feed.

Next, I’ll grab a few sample Medium blog RSS feeds that have actual code blocks in them. Then I’ll compare those with what gets pulled by the aggregator here, just to see where things are breaking or changing.

After that, I’ll get into the "grab.py" script and related docs in the repo. I want to check out exactly how we’re handling iframe and code blocks in the RSS at the moment.

I’ll run some local tests to watch how RSS parsing is behaving. My goal is to see if those code snippets are getting ignored, if the formatting is off, or what’s happening under the hood.

Based on that, I’ll put together a solution. Maybe that means preserving code blocks using HTML/Markdown, fixing parsing logic for iframes, or improving the RSS data handling—will decide based on what I find.

I’ll also include a flowchart right here in the comment to visually explain my step-by-step plan, so it’s easy to see how I intend to tackle it.

My plan is to post this detailed approach along with the diagram and wait for your feedback. Once you’re on board or assign it, I’ll get started with implementation and happy to adjust the plan if you have further suggestions.

Let me know if you need any other information or want me to share some sample data. Thanks a lot for your guidance—I’m looking forward to contributing.

Here’s my roadmap diagram for the contribution steps:

<img width="1301" height="703" alt="Image" src="https://github.com/user-attachments/assets/4d3a29de-bf5f-47ed-bb17-effd14489031" />

**Comment by dpshelio** [comment]
Awesome!! Go for it, I don't have any sample data, but you can find from the repository who was using medium, and which posts had code snippets. I haven't looked into that in detail, but if they are not provided in the feed (or at least signaled that a code snippet is there), then it would be annoying as we would have to check the source to find out for all.

**Comment by aditya-pandey-dev** [comment]
Thank you for your quick response and the clear guidance! I really appreciate the opportunity to work on this issue. I'll start by collecting a few Medium blog posts that use code snippets, and then compare how they're parsed with the current setup here. If I spot any missing or unmarked code blocks in the RSS feed, I'll be sure to check the original sources, just as you suggested.  I'll share what I find and reach out if I need any help or feedback along the way. Excited to get started—thanks again for your support.

**Comment by aditya-pandey-dev** [comment]
Heyy @dpshelio 
Can you please share the GitHub repository link for the code block parsing, scraping, or analysis ?

I need to do some debugging, so could you please send me the repository URL?

**Comment by nabobalis** [comment]
Is it not this repository?

**Comment by aditya-pandey-dev** [comment]
Yes, thanks! I was just double-checking to be sure, since sometimes related logic is in another sub-repo/library.
I will start exploring the code in this repository (especially grab.py and relevant scripts) for the code block/RSS parsing logic.
If I need any more specific pointers, I’ll post here. Thanks for confirming.

**Comment by aditya-pandey-dev** [comment]
Hey @dpshelio I have worked on this issue, please go through the PR: https://github.com/OpenAstronomy/Universe_OA/pull/103
